### ISSUE TITLE
Warn on Always True if expression for tuples

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#4071]
**Is your feature request related to a problem? Please describe.**
```python
def foo() -> tuple[bool, str]:
  ...

if foo():
  ...
```

is accepted but suspicious because the if expression will always be true because it's non-empty tuple. I recently hit a bug like this where someone added a validation message (sorta like pattern for asserts) and caused a very different place of code to behave wrong.

An even stricter rule of only allow bools for ifs I'd probably use, but I realize it's common pattern to do if seq: ... and rely on empty sequences being false even though I tend to use len(seq) == 0 to be cautious of getting behavior confused.

**Describe the solution you'd like**
Warning on if expression values that are always true based on type. I picked fixed length tuple as special case that should be pretty safe, but other types that are always true/false would be nice.

**Comment by erictraut** [comment]
I don't think this is common enough of a bug-generating coding pattern to justify implementing a special check for it. I think it also has potential for a false positive, especially when the conditional involves a call expression because calls can have stateful side effects.

I'm going to close this for now, but we might reconsider in the future if there it receives additional interest.
